### ISSUE TITLE
Error in `rownames<-(*tmp*, value = logical(0))`

### ISSUE THREAD

**Issue by lczech** [petrov-lab/HAFpipe-line#3]
Hi there,

I was trying to get HAFpipe to run on my Arabidopsis data, using this script:

```
#!/bin/bash

cd "$(dirname $0)/"
BASEDIR=`pwd`
export PATH="${BASEDIR}/harp_linux_140925_103521/bin:$PATH"
export PATH="${BASEDIR}/HAFpipe-line-master:$PATH"
HAFPIPE="${BASEDIR}/HAFpipe-line-master/HAFpipe_wrapper.sh"

OUTDIR="out"
mkdir -p ${OUTDIR}

# cd HAFpipe-line-master

FOUNDER="/home/lucas/Projects/1001g/1001gbi.recode.vcf.gz"
SUBSET="/home/lucas/Projects/grenedalf-paper/haf-pipe/founders.txt"

REF="//home/lucas/Projects/grenedalf-paper/reference/TAIR10_chr_all.fa"
BAM="/home/lucas/Projects/grenedalf-paper/benchmark-real/mapped/S1-1.sorted.bam"

${HAFPIPE} \
    -t 1,2,3,4 \
    -v ${FOUNDER} \
    -u ${SUBSET} \
    -c 1 \
    -s ${OUTDIR}/founderGenotypes.segregating.snpTable \
    -i simpute \
    -b ${BAM} \
    -e sanger \
    -g 2     \
    -r ${OUTDIR}/dmel_ref_r5.39.fa \
    -o ${OUTDIR}
```

which produced this log file:

```
        Tue 08 Jun 2021 03:57:35 PM PDT
        ####PARAMETERS########
        --tasks: 1
        --scriptdir /home/lucas/Projects/grenedalf-paper/haf-pipe/HAFpipe-line-master
        --outdir out
        --vcf /home/lucas/Projects/1001g/1001gbi.recode.vcf.gz
        --chrom 1
        --snptable out/founderGenotypes.segregating.snpTable
        --mincalls 2
        
        --subsetlist --subsetlist /home/lucas/Projects/grenedalf-paper/haf-pipe/founders.txt
        --impmethod .simpute
        --nsites 20
        --bamfile /home/lucas/Projects/grenedalf-paper/benchmark-real/mapped/S1-1.sorted.bam
        --refseq out/dmel_ref_r5.39.fa
        --encoding sanger
        --generations 2
        --recombrate .0000000239
        --quantile 18
        --winsize 

        #####COMMANDS#######




COMMAND: /home/lucas/Projects/grenedalf-paper/haf-pipe/HAFpipe-line-master/make_SNPtable_from_vcf.sh -v /home/lucas/Projects/1001g/1001gbi.recode.vcf.gz -c 1 -s out/founderGenotypes.segregating.snpTable --mincalls 2 --subsetlist /home/lucas/Projects/grenedalf-paper/haf-pipe/founders.txt 
making snptable for chrom 1 from /home/lucas/Projects/1001g/1001gbi.recode.vcf.gz, starting from column 10
subsetting snp file out/founderGenotypes.segregating.snpTable to 50 selected founders and writing to out/founderGenotypes.segregating.snpTable.subset
50  fields extracted,  50  fields and 0 sites in new SNP file:
        out/founderGenotypes.segregating.snpTable.subset
allele counts written to:
out/founderGenotypes.segregating.snpTable.alleleCts
making numeric version of out/founderGenotypes.segregating.snpTable
bgzipping out/founderGenotypes.segregating.snpTable.numeric
SNP table written to:
out/founderGenotypes.segregating.snpTable
COMMAND: /home/lucas/Projects/grenedalf-paper/haf-pipe/HAFpipe-line-master/impute_SNPtable.sh out/founderGenotypes.segregating.snpTable
counting alleles in 
```

(the last line seems to have an empty file name printed!)

but then stopped with these errors:

```
./run.sh 
Error in `rownames<-`(`*tmp*`, value = logical(0)) : 
  attempt to set 'rownames' on an object with no dimensions
Execution halted
tail: cannot open 'out/founderGenotypes.segregating.snpTable.numeric' for reading: No such file or directory
```

At this point, it just hangs and does not continue any more.

Here are the files produced so far: [out.zip](https://github.com/petrov-lab/HAFpipe-line/files/6620307/out.zip)

Any idea what is going on here?
Thanks for the support and all the best
Lucas

**Comment by zajacn** [comment]
Hi Lucas,
I had the same error coming up. For me the problem was the lack of the directory I was in in my path (thus wrong search locations) and an active base conda environment still pointing to an old version of R even when I loaded the newest version (so the newest version was not used where I had the dependencies such as data.table installed). I was using R (4.1.2) and Python 2.7.  
export PATH=$PATH:. as well as deactivating the base conda environment solved the issue.
All the best,
Natalia

**Comment by lczech** [comment]
Thank you!

**Comment by lczech** [comment]
Leaving this here for future reference, in case any one has the same error:

Update after having been working more with HAF-pipe: It seems likely that my original issue was caused by #7 instead. That is, the computation of the `numeric` SNP table ran out of memory, and hence that file was never created, which would explain the error as seen above. This is likely, because the input founder VCF for the above run was rather large.

Edit: This whole issue, including #7, is fixed in my PR #9, which, once merged, will solve this issue here as well :-)
